extensions/sockets/src/main/java/org/mule/module/socket/api/provider/TcpListenerProvider.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.provider;
+
+import org.mule.module.socket.api.client.TcpListenerClient;
+import org.mule.module.socket.api.config.ListenerConfig;
+import org.mule.module.socket.api.protocol.SafeProtocol;
+import org.mule.module.socket.api.protocol.TcpProtocol;
+import org.mule.module.socket.api.tcp.TcpServerSocketProperties;
+import org.mule.module.socket.internal.ConnectionSettings;
+import org.mule.module.socket.internal.DefaultTcpServerSocketProperties;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionExceptionCode;
+import org.mule.runtime.api.connection.ConnectionHandlingStrategy;
+import org.mule.runtime.api.connection.ConnectionHandlingStrategyFactory;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+import org.mule.runtime.extension.api.annotation.Alias;
+import org.mule.runtime.extension.api.annotation.Parameter;
+import org.mule.runtime.extension.api.annotation.ParameterGroup;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+
+@Alias("tcp-listener")
+public class TcpListenerProvider implements ConnectionProvider<ListenerConfig, TcpListenerClient>
```

Comment:
wrong usage of the ConnectionProvider all over... You should only have two connection providers. One for TCP and another for UDP. Having one per protocol and per config is not only a violation to the model, it will also prevent you from using the socket config as a global configuration template which is what HTTP uses it for.

The entire connectivity piece is wrong